src/domain/LottoPrize.js

```diff
@@ -0,0 +1,22 @@
+export default class LottoPrize {
+  requiredMatchCount;
+  bonusMatched;
+  prizeMoney;
+  matchCount = 0;
+
+  constructor(requiredMatchCount, bonusMatched, prizeMoney) {
+    this.requiredMatchCount = requiredMatchCount;
+    this.bonusMatched = bonusMatched;
+    this.prizeMoney = prizeMoney;
+  }
+
+  checkPrizeMatch(results) {
+    results.forEach(({ matchCount, bonusMatched }) => {
+      const matched =
+        this.requiredMatchCount === matchCount &&
+        (this.bonusMatched ? this.bonusMatched === bonusMatched : true);
+
+      if (matched) this.matchCount = +1;
```

Comment:
이러면 매번 matchCount의 값이 1로 초기화 될거 같은데 아래 코드를 의도하신걸까요?
```suggestion
      if (matched) this.matchCount += 1;
```